### PR TITLE
Merge Practice app/feature/discussion_comments into practice_app

### PR DESCRIPTION
**Pull request by azizamankenova**
* Tracking of this pull request can be seen in https://github.com/bounswe/bounswe2022group7/issues/163.

* The feature of viewing discussing post with comments and adding new comments is fully implemented, including frontend, backend, unittests, and documentation 

Awaiting review from @canatakan

### CODE REVIEW

**Comment by azizamankenova** [comment]
> CI workflow check seems to have failed.
> 
> Apparently, your unit tests are not okay. You can try your unit tests with `python3 -m unittest` command while you are at the root directory, namely `practice-app`.
> 
> Could you please check your pull request again? @azizamankenova

@canatakan Yes, I have tried running the same command in local and all the tests have passed successfully. Unit tests must have failed because of the API Key, which was not yet provided.

**Review by canatakan — CHANGES_REQUESTED** [review]
Except from minor changes, all seems to be good. Please check my comments, and you can update the code afterwards.

File: practice-app/website/templates/forum_post.html [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            }
+        })
+        .then(response => {
+            if (response.status == 201) {
+                console.log(response.json())
+                location.reload();
+            }   
+        })
+        
+        // .then(response => response.json())
```
[the remark this answers is not in the document]
Comment (azizamankenova) [reply practice-app/website/templates/forum_post.html]: fixed

File: practice-app/website/templates/base.html [hunk 2]
```diff
@@ -65,6 +65,7 @@ <h1>
                 {% if session["access_token"] %}
                     <a href={{ url_for("views.view_profile") }}>View Profile</a>
                 {% endif %}
+            <!-- </div> -->
```
[the remark this answers is not in the document]
Comment (azizamankenova) [reply practice-app/website/templates/base.html]: fixed it

File: practice-app/website/api/discussion_forum.py [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
+from flask_sqlalchemy import SQLAlchemy
+import json
 import requests
+from ..settings import *
+from .jwt import artist_required, user_required
+from http.client import INTERNAL_SERVER_ERROR
+# from website import db
 from flask_jwt_extended import jwt_required, current_user
-from website.settings import *
+# from website.settings import *
```
[the remark this answers is not in the document]
Comment (azizamankenova) [reply practice-app/website/api/discussion_forum.py]: fixed it

File: practice-app/website/api/discussion_forum.py [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
-from website.models import ForumPost
+from .. import db
+from ..models import ForumPost, PostComment, User
+from flask_sqlalchemy import SQLAlchemy
+import json
 import requests
+from ..settings import *
+from .jwt import artist_required, user_required
+from http.client import INTERNAL_SERVER_ERROR
+# from website import db
```
[the remark this answers is not in the document]
Comment (azizamankenova) [reply practice-app/website/api/discussion_forum.py]: fixed it

**Comment by azizamankenova** [comment]
Thanks @canatakan for the review. Fixed everything. Ready to be merged

**Comment by canatakan** [comment]
> Thanks @canatakan for the review. Fixed everything. Ready to be merged

Thanks for your updates! Everything seems fine now, merging.

**Pull request merged**